leaf-ui/js/displayAnalysis.js

```diff
@@ -346,19 +351,44 @@ function addAnalysisConfig(config) {
     mainElement = document.getElementById(config.id);
     mainElement.querySelector('.config-elements').addEventListener('dblclick', function(){rename(this /** Config element */)});
     mainElement.querySelector('.config-elements').addEventListener('click', function(){switchConfigs(this.parentElement /** Config element */)});
-    mainElement.querySelector('.dropdown-button').addEventListener('click', function(){toggleDropdown(this.parentElement.parentElement /** Config element */)})
+    mainElement.querySelector('.dropdown-button').addEventListener('click', function(){toggleDropdown(this.parentElement.parentElement /** Config element */)});
+    mainElement.querySelector('.deleteconfig-button').addEventListener('click', function(){ 
+        try{removeConfiguration(this.parentElement.parentElement)/** Config element */} 
+        catch(e){console.error(e); alert(e);}
+    });
+
+    //Whenever first or loaded configs are added, update the highest position
+    highestPosition = config.initialPosition;
 }
 
 /**
  * Removes an analysis configuration from the analysisMap and UI sidebar
  * TODO implement with a button
+ * @param {HTML Element} configElement
  */
-function removeConfiguration() {
+function removeConfiguration(configElement) {
     // Remove full configuration div (includes results)
-    var configDiv = document.getElementById(currAnalysisConfig.id);
+    var configDiv = document.getElementById(configElement.id);
+    //returns a NodeList (not an array) of all configuration elements
+    var configList = document.querySelectorAll('.analysis-configuration');
```

Comment:
Instead of iterating through this list it might be easier to just use the jquery prev() and next() functions to get the id of the new currAnalysisConfig. You would just need to basically try "prev()" on the configElement first, and then if that returned null use "next()" instead. This would condense lines 373-387 into just two if statements, one nested inside the other which I think would be more concise and easier to follow. [Here is some documentation](https://www.w3schools.com/jquery/traversing_prev.asp) for prev(), next() is pretty much the same. You would just need to use 'div' as the selector too so it would be pretty simple.